### DISCUSSION TITLE
Consider publishing only esm with v17

### DISCUSSION THREAD

**Discussion by yaacovCR** [graphql/graphql-js#3358] [General]
https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#gistcomment-3850849

**Comment by yaacovCR** [comment]
https://github.com/graphql/graphql-js/issues/2740

**Comment by yaacovCR** [comment]
Cf https://github.com/graphql/graphql-js/pull/3217#issuecomment-955876515

**Comment by saihaj** [comment]
Attempted in #3361

**Reply by yaacovCR** [reply]
Actually, I meant even more radical, ie to abandon cjs

**Reply by yaacovCR** [reply]
Another take: https://gist.github.com/joepie91/bca2fda868c1e8b2c2caf76af7dfcad3

**Comment by yaacovCR** [comment]
Note also https://github.com/jaydenseric/graphql-upload/issues/305#issuecomment-1134736158

in terms of only using deep imports for each module (where module is defined essentially as each individually accessible bit of the public API?)

this is a different solution @IvanGoncharov  to always remembering to update the main index file, ie removing it…

**Comment by n1ru4l** [comment]
related: https://github.com/graphql/graphql-js/issues/3603
